pkg/query/storeset.go

```diff
@@ -341,16 +344,30 @@ func (s *StoreSet) updateStoreStatus(store *storeRef, err error) {
 	s.storesStatusesMtx.Lock()
 	defer s.storesStatusesMtx.Unlock()
 
+	var status StoreStatus
 	now := time.Now()
-	s.storeStatuses[store.addr] = &StoreStatus{
-		Name:      store.addr,
-		LastError: err,
-		LastCheck: now,
-		Labels:    store.labels,
-		StoreType: store.storeType,
-		MinTime:   store.minTime,
-		MaxTime:   store.maxTime,
+
+	if previousStatus, ok := s.storeStatuses[store.addr]; ok {
+		status = *previousStatus
+		status.LastError = err
+		status.LastCheck = now
+	} else {
+		status = StoreStatus{
+			Name:      store.addr,
+			LastError: err,
```

Comment:
This bit repeats, so I would just do it once after loop. 

Also I would do it without `if xxx, ok := ... : ok  {` just:

```
status = StoreStatus{Name: store.addr}
prev, ok := ... 
if ok {
status = *prev
}
status.LastError = err 
status.LastCheck = now
```

What do you think?

Reply:
Much simpler yes !